doc: document how to apply a token with github-nodejs-bot (#538)
nodejs/github-bot: APPLY_ACCESS_TOKEN.md

# Apply for a nodejs-github-bot token

> Automation in https://github.com/nodejs may require github access token to access

Reviewer: ```suggestion
Automation in the `nodejs` GitHub Org may require access tokens to access
```

Author: Thanks, updated!

permission scoped endpoints. In the case of such requirement, the access token can

> be requested to be created under the name of https://github.com/nodejs-github-bot. 

Reviewer: ```suggestion
be requested to be created under the name of [`@nodejs-github-bot`][].
```

> Only fine-grained access token can be created under the name of https://github.com/nodejs-github-bot.

Reviewer: ```suggestion
Creating classic tokens for [`@nodejs-github-bot`][] is not permitted, only fine-grained tokens are allowed.
```

> To create a fine-grained access token with nodejs-github-bot, submit a PR to

Reviewer: ```suggestion
To create a fine-grained access token for [`@nodejs-github-bot`][], submit a PR to
```

add the requested repo in the registry below, and a TSC member or a build WG member
(who has access to the https://github.com/nodejs-github-bot account) needs to take

> action to create the fine-grained token and save the token to the requested repo's

Reviewer: We should document the precise process for how to land such PR, create the secret, add it as a repo secret.

> action to create the fine-grained token and save the token to the requested repo's
> repository secrets.

Reviewer: ```suggestion
action to create the fine-grained token and save the token as a repository secret.
```

Fine-grained tokens created with access to https://github.com/nodejs resources will
be audited at https://github.com/organizations/nodejs/settings/personal-access-tokens/active.

## Registry

The "repo" is a string of the GitHub `<owner>/<repo>`. Generally, the token should
only be created for repo in the https://github.com/nodejs organization.

> The "token name" is a string that the token can be referenced in the GitHub Action
> scripts. Like a token name of `RELEASE_PLEASE_TOKEN` can be accessed from the script
> as `${{ secrets.RELEASE_PLEASE_TOKEN }}`.
> 
> Repo                        | Token name

Reviewer: ```suggestion
The "Secret name" is a string that the token can be referenced in the GitHub Action
scripts. Like a token name of `RELEASE_PLEASE_TOKEN` can be accessed from the script
as `${{ secrets.RELEASE_PLEASE_TOKEN }}`.

Repo                        | Secret name
```
